sys/include/net/gnrc/netif.h

```diff
@@ -565,6 +565,19 @@ int gnrc_netif_get_from_netdev(gnrc_netif_t *netif, gnrc_netapi_opt_t *opt);
 int gnrc_netif_set_from_netdev(gnrc_netif_t *netif,
                                const gnrc_netapi_opt_t *opt);
 
+/**
+ * @brief Search the list of netifs for a netdev of a certain type.
+ *
+ * @pre The netdev has been registered with @ref netdev_register
+ *
+ * @param[in]  type         driver type of the netdev
+ * @param[in]  index        index of the netdev
+ *
+ * @return  The netdev that matches the type and index
+ *          NULL if no matching netdev could be found
+ */
+netdev_t *gnrc_netif_get_netdev(netdev_type_t type, uint8_t index);
```

Comment:
I have some concerns with this functions:

1) As it is, it collides with the ongoing efforts for removing the `netdev` dependency of GNRC netif (#13614 )
2) By design GNRC netif has full control of the netdev interface. It seems to me that this function is there for allowing direct access to the `netdev` descriptor. If we allow that without passing through `gnrc_netif` via NETOPTs, we will eventually end up with corrupted radio states and thread safety issues.

Are there any other reasons for this function, that I might have missed?

I agree though, that we have no mechanism for getting a device descriptor by ID or type and I think we should definitely aim for that.
I heard this during offline conversations with @MichelRottleuthner and @haukepetersen. Maybe we should start that topic again

Reply:
Should we really add driver specific NETOPTs though?

Then again, maybe this is the right approach to take. Bu instead of further bloating the `ifconfig` command, I would add an `at86rf215` command that would then issue NETOPTs.

Reply:
hmmmm maybe a NETOPT for accessing private stuff? (NETOPT_PRIVATE).
I know it might sound funny, but we are already at the point where the NETOPT list is totally bloated with device/class dependent stuff.

I was already thinking of doing something like NETOPTs grouping.
- `NETOPT_LINK_LORA`
    - `LORAWAN_SF`
    - `LORAWAN_BW`

Then, the `netopt.h` only defines the primitives (`NETOPT_LINK_LORA` `NETOPT_PRIVATE`, etc). Then the sub groups can be defined in other files (e.g there could be an `at86rf215_opt_t` enum with these stuff).

What do you think?

Reply:
I'd go with `NETOPT_AT86RF215_`.

Btw.: Channel Bandwidth is not limited to LoRa, we can also set this with 802.15.4g. I'd like to pull that out of the driver (together with channel spacing) in the future, so an upper layer can keep track of which which bandwidth / center frequency combinations are allowed in different world regions.

Reply:
How about `gnrc_netif_get_by_type()` (in the same spirit as the other `gnrc_netif_get_by_` functions)?

Reply:
that sounds like a clean approach to me